versioned_hdf5/api.py

```diff
@@ -230,3 +240,69 @@ def __repr__(self):
         else:
             return f"<VersionedHDF5File object \"{self.f.filename}\" (mode" \
                    f" {self.f.mode})>"
+
+    def get_hashes(self, name: str) -> Set[bytes]:
+        """Get a set of hashes for the chunks in the dataset.
+
+        Parameters
+        ----------
+        name : str
+            Name of the dataset for which hashes are to be generated
+
+        Returns
+        -------
+        Set[bytes]
+            A set of hashes for the dataset
+        """
+        with Hashtable(self.f, name) as hashtable:
+            return set(hashtable.keys())
+
+    def _log_version_diff_stats(
+        self,
+        old_version: Optional[str] = None,
+        new_version: Optional[str] = None,
+    ):
+        """Log some stats about differences between two versions.
+
+        Parameters
+        ----------
+        old_version : Optional[str]
+            Old version of the data to compare
+        new_version : Optional[str]
+            New version of the data to compare
+        """
+        old_datasets, new_datasets = {}, {}
+        if old_version in self:
+            old_datasets = self[old_version].datasets()
+        if new_version in self:
+            new_datasets = self[new_version].datasets()
+
+        msg = [""]
+        for name in sorted(set(old_datasets.keys()) | set(new_datasets.keys())):
+            old_dataset = old_datasets.get(name, None)
+            new_dataset = new_datasets.get(name, None)
+
+            old_hashes, new_hashes = set(), set()
+            old_shape, new_shape = None, None
+            old_chunks, new_chunks = None, None
+
+            if old_dataset:
+                old_shape = old_dataset.shape
+                old_chunks = old_dataset.chunks
+                old_hashes = self.get_hashes(name)
+
+            if new_dataset:
+                new_shape = new_dataset.shape
+                new_chunks = new_dataset.chunks
+                new_hashes = self.get_hashes(name)
+
+            chunks_reused = len(old_hashes & new_hashes)
+            new_chunks_written = len(new_hashes - old_hashes)
+
+            msg.append(
+                f"  {name}: Shape: {old_shape} -> {new_shape}; "
+                f"Chunks: {old_chunks} -> {new_chunks}; "
+                f"New chunks written: {new_chunks_written}; "
+                f"Number of chunks reused: {chunks_reused}"
+            )
+        logger.info("\n".join(msg))
```

Comment:
This should log at debug level by default, I think. Most of our production jobs run with info level debugging and would end up printing this, which our end users probably do not care about.

Reply:
Sounds good to me.